internal/pkg/mount/system.go

```diff
@@ -195,13 +199,19 @@ func SystemPartitionMount(r runtime.Runtime, label string, opts ...Option) (err
 		return err
 	}
 
+	mountpointsMutex.Lock()
+	defer mountpointsMutex.Unlock()
+
 	mountpoints[label] = mountpoint
 
 	return nil
 }
 
 // SystemPartitionUnmount unmounts a system partition by the label.
 func SystemPartitionUnmount(r runtime.Runtime, label string) (err error) {
+	mountpointsMutex.Lock()
+	defer mountpointsMutex.Unlock()
+
 	if mountpoint, ok := mountpoints[label]; ok {
 		err = mountpoint.Unmount()
```

Comment:
I think we better use more granular lock there to avoid wrapping (potentially long) Umount() call with lock.

Even better, we could use RWMutex.

Reply:
great point, I think we could read mountpoint, and unlock mutex? we don't need to hold the mutex for the duration of the mount

Reply:
Exactly. We can wrap `mountpoints` reads with RLock / RUnlock, and `delete` below (and assignment above) with Lock / Unlock

Reply:
sure, fixed.